tensordict/nn/distributions/composite.py

```diff
@@ -147,19 +166,91 @@ def rsample(self, shape=None) -> TensorDictBase:
             shape + self.batch_shape,
         )
 
-    def log_prob(self, sample: TensorDictBase) -> TensorDictBase:
-        """Writes a ``<sample>_log_prob entry`` for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob."""
+    def log_prob(self, sample: TensorDictBase) -> torch.Tensor | TensorDictBase:
+        """Computes and returns the summed log-prob.
+
+        If ``self.aggregate_probabilities`` is ``True``, this method will return a single tensor with
+        the summed log-probabilities. If ``self.aggregate_probabilities`` is ``False``, this method will
+        call the `:meth:`~.log_prob_composite` method and return a tensordict with the log-probabilities
+        of each sample in the input tensordict along with a ``sample_log_prob`` entry with the summed
+        log-prob. In both cases, the output shape will be the shape of the input tensordict.
+        """
+        if not self.aggregate_probabilities:
+            return self.log_prob_composite(sample, include_sum=True)
+        slp = 0.0
+        for name, dist in self.dists.items():
+            lp = dist.log_prob(sample.get(name))
+            while lp.ndim > sample.ndim:
+                lp = lp.sum(-1)
+            slp = slp + lp
+        return slp
+
+    def log_prob_composite(
+        self, sample: TensorDictBase, include_sum=True
+    ) -> TensorDictBase:
+        """Writes a ``<sample>_log_prob`` entry for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob.
+
+        This method is called by the :meth:`~.log_prob` method when ``self.aggregate_probabilities`` is ``False``.
+        """
         slp = 0.0
         d = {}
         for name, dist in self.dists.items():
             d[_add_suffix(name, "_log_prob")] = lp = dist.log_prob(sample.get(name))
             while lp.ndim > sample.ndim:
                 lp = lp.sum(-1)
             slp = slp + lp
-        d[self.log_prob_key] = slp
+        if include_sum:
+            d[self.log_prob_key] = slp
         sample.update(d)
         return sample
 
+    def entropy(self, samples_mc=1) -> torch.Tensor | TensorDictBase:
+        """Computes and returns the summed entropies.
+
+        If ``self.aggregate_probabilities`` is ``True``, this method will return a single tensor with
+        the summed entropies. If ``self.aggregate_probabilities`` is ``False``, this method will call
+        the `:meth:`~.entropy_composite` method and return a tensordict with the entropies of each sample
+        in the input tensordict along with an ``entropy`` entry with the summed entropy. In both cases,
+        the output shape will match the shape of the distribution sample, without a batch dimension.
```

Comment:
isn't it the opposite, we only have the batch dimension, no feature dim?

Reply:
So, for a single distribution, the entropy method returns a tensor with the shape of the distribution samples. 
In this case we have one tensor for each of the distributions of the composite distribution. 
Each output can have different shapes.
What we can do is to assume all outputs are independent and sum all entropies, returning then a tensor of size `torch.Size([])`.
And to retrieve the entropies of individual dists with their specific shape `composite_entropy` can be used.